File: 0047-permutations-ii/0047-permutations-ii.cpp
class Solution {
public:
     vector<vector<int>>v;
   
    void solve(vector<int>nums,int ind,int n)
    {
        if(n==ind ){
           if(find(v.begin(),v.end(),nums)==v.end()) v.push_back(nums);return;
        }
            
        for(int i=ind;i<n;i++)
        {
            swap(nums[i],nums[ind]);
            solve(nums,ind+1,n);    
        }
        
        
    }
    vector<vector<int>> permuteUnique(vector<int>& nums) {
       int n=nums.size();
        
        solve(nums,0,n);
        return v;
    }
};

File: 0047-permutations-ii/0047-permutations-ii.java

class Solution {
    List<List<Integer>> ans;

    public void solve(int[] nums, List<Integer> cur, boolean[] vis) {
        
        if (cur.size() == nums.length) {
            ans.add(new ArrayList<>(cur));
            return;
        }

        for (int i = 0; i < nums.length; i++) {
            if (vis[i]) continue;
                
            if (i > 0 && nums[i] == nums[i - 1] && !vis[i - 1]) continue;

            vis[i] = true;
            cur.add(nums[i]);      
            
            solve(nums, cur, vis); 
            
            cur.remove(cur.size() - 1);
            vis[i] = false; 
        }
    }
    
    public List<List<Integer>> permuteUnique(int[] nums) {
        List<Integer> cur = new ArrayList<>();
        Arrays.sort(nums); 
        ans = new ArrayList<>();
        boolean[] vis = new boolean[nums.length];
        solve(nums, cur, vis);
        
        return ans;
    }
}